jenkins_jobs/trigger_host_os_iso_build_from_versions_repo/script.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+set -x
+
+
+TARGET_CONTEXT="Build Host OS"
+
+
+get_build_state(){
+    # this will evaluate the queried keys as variables: state=<state>,
+    # target_url=<url>
+    eval $(github_api "$GITHUB_USER_NAME" "$GITHUB_PASSWORD" \
+                      query_status "${GITHUB_ORGANIZATION_NAME}/versions" \
+                      "$ghprbPullId" "$TARGET_CONTEXT" --state \
+                      --target-url || echo "exit 1")
+}
+
+get_pr_state(){
+    # this will evaluate the queried keys as variables: state=<state>,
+    # title=<title>
+    eval $(github_api "$GITHUB_USER_NAME" "$GITHUB_PASSWORD" \
+                      query_pr "${GITHUB_ORGANIZATION_NAME}/versions" \
+                      "$ghprbPullId" --state --title || echo "exit 1")
+}
+
+state="pending"
+get_build_state
+while [ $state == "pending" ]; do
+    echo "Waiting for $TARGET_CONTEXT to finish..."
+    echo "$target_url"
+    sleep 1m
+    get_build_state
+done
+
+if [ "$state" == "failure" -o "$state" == "error" ]; then
+    echo "$TARGET_CONTEXT failed, aborting."
+    echo "$target_url."
+    exit 1
+fi
+
+if [ "$state" == "success" ]; then
+    echo "$TARGET_CONTEXT succeeded."
+    echo "$target_url"
+    get_pr_state
+    if [ "$state" == "closed" ]; then
+        echo "Pull request ${ghprbPullId}: \"${title}\" was closed, aborting."
+        echo "$ghprbPullLink"
+        exit 1
+    fi
+fi
+
+echo "BUILD_JOB_NUMBER=$(basename $target_url)" > BUILD_PARAMETERS
+echo "Triggering ISO build from $target_url ..."
+
+exit 0
```

Comment:
side-note: it seems to me we are writing all the code which coordinates Jenkins jobs in shell script, IMHO we should consider doing it in Python, it would make the code much simpler and make maintenance easier